patches/0004-Output-to-hashed-mappings.patch

```diff
@@ -0,0 +1,1622 @@
+ 		TinyRemapper remapper = TinyRemapperHelper.getTinyRemapper(project, fromM, toM, true, (builder) -> {
+ 			builder.extraPostApplyVisitor(new SignatureFixerApplyVisitor(remappedSignatures));
+ 			configureRemapper(remappedJars, builder);
+diff --git a/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/HashedMinecraftProvider.java b/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/HashedMinecraftProvider.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..71f7159df06f5f300fa28920194e535ecfec5195
+--- /dev/null
++++ b/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/HashedMinecraftProvider.java
+@@ -0,0 +1,117 @@
++/*
++ * This file is part of fabric-loom, licensed under the MIT License (MIT).
++ *
++ * Copyright (c) 2022 FabricMC
++ *
++ * Permission is hereby granted, free of charge, to any person obtaining a copy
++ * of this software and associated documentation files (the "Software"), to deal
++ * in the Software without restriction, including without limitation the rights
++ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
++ * copies of the Software, and to permit persons to whom the Software is
++ * furnished to do so, subject to the following conditions:
++ *
++ * The above copyright notice and this permission notice shall be included in all
++ * copies or substantial portions of the Software.
++ *
++ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
++ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
++ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
++ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
++ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
++ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
++ * SOFTWARE.
++ */
++
++package net.fabricmc.loom.configuration.providers.minecraft.mapped;
++
++import java.nio.file.Path;
++import java.util.List;
++
++import org.gradle.api.Project;
++
++import net.fabricmc.loom.api.mappings.layered.MappingsNamespace;
++import net.fabricmc.loom.configuration.providers.minecraft.MergedMinecraftProvider;
++import net.fabricmc.loom.configuration.providers.minecraft.MinecraftProvider;
++import net.fabricmc.loom.configuration.providers.minecraft.SingleJarMinecraftProvider;
++import net.fabricmc.loom.configuration.providers.minecraft.SplitMinecraftProvider;
++import net.fabricmc.loom.util.SidedClassVisitor;
++import net.fabricmc.tinyremapper.TinyRemapper;
++
++public abstract sealed class HashedMinecraftProvider<M extends MinecraftProvider> extends AbstractMappedMinecraftProvider<M> permits HashedMinecraftProvider.MergedImpl, HashedMinecraftProvider.SingleJarImpl, HashedMinecraftProvider.SplitImpl {
```

Comment:
I think in this case what you should do instead is modify the Intermediary minecraft provider to be extendable by hashed, and then only override the methods you actually have to change, since i believe most of this class is duplicated?

Reply:
That would be smarter yes lol, I'll do that now

Reply:
I honestly don't know if this is worth all the changes it would make, it would be hard to make this work with the mess of generics that is `MinecraftJarConfiguration`.